impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultPluginXmlFactory.java

```diff
@@ -38,65 +37,63 @@
 import org.apache.maven.plugin.descriptor.io.PluginDescriptorStaxReader;
 import org.apache.maven.plugin.descriptor.io.PluginDescriptorStaxWriter;
 
-import static org.apache.maven.impl.ImplUtils.nonNull;
 import static org.apache.maven.impl.StaxLocation.getLocation;
 import static org.apache.maven.impl.StaxLocation.getMessage;
 
 @Named
 @Singleton
 public class DefaultPluginXmlFactory implements PluginXmlFactory {
+
     @Override
     public PluginDescriptor read(@Nonnull XmlReaderRequest request) throws XmlReaderException {
-        nonNull(request, "request");
-        Path path = request.getPath();
-        URL url = request.getURL();
-        Reader reader = request.getReader();
-        InputStream inputStream = request.getInputStream();
-        if (path == null && url == null && reader == null && inputStream == null) {
-            throw new IllegalArgumentException("path, url, reader or inputStream must be non null");
-        }
+        return read(request, setAddDefaultEntities(request, new PluginDescriptorStaxReader()));
+    }
+
+    @Override
+    public void write(XmlWriterRequest<PluginDescriptor> request) throws XmlWriterException {
```

Comment:
goofy, having a [factory](https://refactoring.guru/design-patterns/creational-patterns) `not` creating anything:

<img width="726" alt="image" src="https://github.com/user-attachments/assets/aeb28a22-0e80-4426-ba04-4be8ec22d20b" />

Reply:
Agreed, but in that case, the PR is off-topic and maybe the point is to rename the interfaces.